### DISCUSSION TITLE
Bluetooth client vs server mode

### DISCUSSION THREAD

**Discussion by Dalowend** [Builty/TonexOneController#24] [Q&A] [answered]
Hi, can you please explain the difference between Bluetooth client vs server modes? What are the different configurations that I should think about?

**Comment by Builty** [comment, marked as answer]
Hi. In the Bluetooth world, there are 2 main types of devices (and some rarer other ones)
- Peripherals. These are devices like a keyboard or mouse. They "advertise" that they are available for connection, and if something connects, then things like key presses can be reported
- Centrals. These are "host" devices, like a PC or a phone. They don't usually advertise. They listen for advertising from peripherals, and can connect to them

So for this Tonex controller, "client" mode is another way of describing a Central device. In Client mode, the controller will go looking for peripherals, like a Midi footswitch, and connect to them and receive Midi data.

Server mode is the opposite. The controller becomes a peripheral. You can these use a PC or phone or similar, and see the controller as available. You can then connect the PC/phone to the controller, and send Midi data to change presets.

In summary, if you want to connect a Midi controller device to it, use Client mode.
If you want to connect the controller to a PC/Phone, use Server mode

**Reply by Dalowend** [reply]
that explains it, thanks!
